src/app/modules/item/components/chapter-children/chapter-children.component.ts

```diff
@@ -24,18 +26,47 @@ interface ItemChildAdditions {
 export class ChapterChildrenComponent implements OnChanges, OnDestroy {
   @Input() itemData?: ItemData;
 
-  state: 'loading' | 'error' | 'empty' | 'ready' | 'ready-missing-validation' = 'loading';
-  children: (ItemChild&ItemChildAdditions)[] = [];
+  private readonly params$ = new ReplaySubject<{ id: string, attemptId: string }>(1);
+  readonly state$ = this.params$.pipe(
+    distinctUntilChanged((a, b) => a.id === b.id && a.attemptId === b.attemptId),
+    switchMap(({ id, attemptId }) =>
+      this.getItemChildrenService.get(id, attemptId).pipe(
+        map((itemChildren: ItemChild[]) => {
```

Comment:
Unless I'm missing something, there's no need to nest the `map` operator into the `switchMap`
When possible, we should keep observable declarations as flat as possible for readability

Plus, considering the size of the `map`, I think some of its content can be externalised into class methods